### ISSUE TITLE
Attempting to export to PhoneGap hangs on Uploading Project...

### ISSUE THREAD

**Issue by simsketch** [ekelokorpi/panda-engine#204]
**Describe the bug**
Attempting to export to PhoneGap hangs on Uploading Project...

**To Reproduce**
1. Go to Export
2. Click on APK
3. Fill out keystore etc.
4. PhoneGap login is successful
4. Gets to Uploading Project... and never completes, even after leaving overnight. I need to quit in order to use the app again. If there is an error popping up somewhere, I wouldn't know where to look.

**Expected behavior**
A successful upload to my PhoneGap build account

**Screenshots**
![image](https://user-images.githubusercontent.com/3129129/46149794-166d8200-c239-11e8-843c-672700bb6599.png)

**Info**
 - OS: macOS High Sierra 10.13.6
 - Version: Panda Engine 2.9.0

**Comment by ekelokorpi** [comment]
Thanks for the report!

What editor version are you using?

Can you send your log file located at ~/Library/Logs/Panda 2/log.log

**Comment by simsketch** [comment]
Thank you! (show hidden files with command + shift +  . ) I see the error says "Private app limit reached" which is weird since i only have one app ...but it's definitely not a Panda thing! Closing...

**Comment by simsketch** [comment]
Ok now it gets to the next step, tells me Keystore file not found.

I am using the following commands to generate my keystore.

```
keytool -genkey -alias myKey -keystore store.jks
keytool -selfcert -alias myKey -keystore store.jks
```

any suggestions? thanks.

**Comment by ekelokorpi** [comment]
Have you placed the keystore file in the root of your project?

**Comment by simsketch** [comment]
ok i tried placing the keystore in the root and referenced with an absolute path but i still got `Keystore file not found`

![image](https://user-images.githubusercontent.com/3129129/46160493-db774880-c250-11e8-931d-63bfdf5801b9.png)

after trying it with a relative path the dev console says Building apk...no error this time...looks like it was successful, but i don't see the apk in the root folder if that's where it's supposed to be saved.
![image](https://user-images.githubusercontent.com/3129129/46160705-6a846080-c251-11e8-9b50-8c463003d84e.png)
![image](https://user-images.githubusercontent.com/3129129/46160717-71ab6e80-c251-11e8-90e5-1b2f730eaab1.png)

**Comment by ekelokorpi** [comment]
When you click on the Build button, it will ask for a folder where the APK will be saved.

**Comment by simsketch** [comment]
ok that's what i thought...when i close the Editor and reopen, attempt to Export again, it says Building in progress...

**Comment by simsketch** [comment]
what would happen if i selected the same folder as the game i'm exporting? is that acceptable? that's what i chose for this "hanging" build

**Comment by simsketch** [comment]
ok i Ended Panda processes in Activity Monitor and was able to attempt another export

**Comment by ekelokorpi** [comment]
Ok so did it worked now?

**Comment by simsketch** [comment]
dev tools says Building APK.... but nothing shows up in the folder i selected...if i try another export it again says `Buliding already in progress` hmm.

**Comment by ekelokorpi** [comment]
What did it say after "Building APK..." ? It can take a while so you should wait.

**Comment by ekelokorpi** [comment]
You are using "Background build"? Can you disable that and try again.

**Comment by simsketch** [comment]
I tried it with Background build disabled and it still hangs on `Building APK` in Panda editor

![image](https://user-images.githubusercontent.com/3129129/46171567-e0e38b80-c26e-11e8-8256-94e8aa4caea1.png)

**Comment by simsketch** [comment]
It ended up staying like that for hours...still going. I tried it on two different mac pros with the same behavior.

**Comment by ekelokorpi** [comment]
This doesn't sound right.

No i did not do any updates on the engine, and i believe this has nothing to do with the engine, sounds more like an issue with the editor.

Can you see the app in your PhoneGap Build apps after it hangs in the "Building APK..."? https://build.phonegap.com/apps

Also if you could send ZIP of your project i can take a closer look and see what's the issue. You can email me at eemeli.kelokorpi@gmail.com

**Comment by simsketch** [comment]
The app does show up in PhoneGap but if I try to download the apk, it has debug in the filename and it doesn't seem to finish the download.

![untitled](https://user-images.githubusercontent.com/3129129/46208396-9eff2780-c2f8-11e8-901b-e4d9268105ff.png)

**Comment by simsketch** [comment]
ok i got it to download successfully from PhoneGap UI but uploading to Google Play failed because it's a debuggable APK

**Comment by ekelokorpi** [comment]
Hi. I just tried to export Android Release build with your project and i did not have any problems.

There might be something not correct in your keystore file. Can you build it using this guide and try again: http://docs.phonegap.com/phonegap-build/signing/android/

Be sure to enter the correct key alias into the export settings in Panda.

For example if i build my keystore with this command:
`keytool -genkey -v -keystore release.keystore -alias cutetitos -keyalg RSA -keysize 2048 -validity 10000`

My Android export settings should look like this:
![screen shot 2018-09-28 at 18 22 26](https://user-images.githubusercontent.com/1164289/46217729-7b190180-c34b-11e8-83d0-a740a0085d9c.png)

**Comment by simsketch** [comment]
Success in building a 1_0_0-release.apk! However, when attempting to upload to Google Play, I get the same "this is a debug apk" error.

> Upload failed
> You uploaded a debuggable APK or Android App Bundle. For security reasons you need to disable debugging before it can be published in Google Play. Learn more about debuggable APKs or Android App Bundles.
> You uploaded an APK or Android App Bundle that was signed in debug mode. You need to sign your APK or Android App Bundle in release mode. Learn more about signing.

I definitely had release build checked, and it did produce the apk labeled release.

**Comment by simsketch** [comment]
Going to try it from the other machine.

**Comment by ekelokorpi** [comment]
Did you define bundle id in your game config and used that same id when creating app in Google Play?

Check out this tutorial and section "Configuring": https://www.panda2.io/tutorials/build

**Comment by simsketch** [comment]
I tried adding id and version in the game config as you suggested...same result.

I tried zipping the cordova folder and uploading that zip with PhoneGap UI, along with the key, unlocked the key, and was able to build an apk. When i tried to upload, I got this error message.

![image](https://user-images.githubusercontent.com/3129129/46227022-fc609c00-c32b-11e8-8d21-a845f21adac3.png)

**Comment by ekelokorpi** [comment]
That is totally different issue. You can set the Android target API in config.xml like this:
`<preference name="android-targetSdkVersion" value="26" />`

The original issue seems to be with PhoneGap Build, there must be something changed. It doesn't seem to build release build even when it is supposed to. Still trying to investigate how to solve the issue.

**Comment by simsketch** [comment]
Thanks...I was able to successfully build a release apk in PhoneGap manually and after adding that line it was accepted by Google Play...thanks!

I'm surprised the `cordova build android` command doesn't add that line automatically.

**Comment by ekelokorpi** [comment]
Finally got this issue fixed. Exporting Android release APK from Panda Editor now works again. Will be fixed in the next 1.6.0 version

**Comment by simsketch** [comment]
wow...awesome...nice work!
